packages/loot-core/src/server/importers/ynab4.ts

```diff
@@ -71,8 +71,29 @@ async function importCategories(
           // on insertion order
           for (const category of subCategories) {
             if (!category.isTombstone) {
+              let categoryName = category.name;
+
+              // Hidden categories have the parent category entity id
+              // appended to the end of the sub category name.
+              // The format is 'MasterCategory ` SubCategory ` entityId'.
+              // Remove the id, the apostrophe and the space before it,
+              // to shorten the name.
+              if (masterCategory.name === 'Hidden Categories') {
+                // Find the last apostrophe in the name.
+                const lastApostrophe = categoryName.lastIndexOf('`');
+                // Remove everything after the last apostrophe, including
+                // it and the space before it.
+                if (lastApostrophe !== -1) {
+                  categoryName = categoryName.substring(0, lastApostrophe - 1);
```

Comment:
@alcroito Being that you want to replace the apostrophe anyways,  I think this might be a bit simpler if we instead split on ' ` ' and just joined with / without the last entry.

Reply:
Thanks, done.